src/uwtools/utils/templater.py

```diff
@@ -12,72 +11,87 @@
-    config_items: List[str],
+    config_file: Optional[str],
+    key_eq_val_pairs: List[str],
     input_template: str,
     outfile: str,
     log: Logger,
     dry_run: bool = False,
     values_needed: bool = False,
 ) -> None:
     """
-    Render a Jinja2 template using user-supplied configuration options via YAML or environment
-    variables.
-    """
-    cfg = _set_up_config_obj(config_file=config_file, config_items=config_items, log=log)
+    Render a Jinja2 template using values fro an input file.
 
-    # Instantiate Jinja2 environment and template.
+    :param config_file: Path to the config file supplying values to render the template.
+    :param keq_eq_val_pairs: "key=value" strings to supplement config-file values.
+    :param input_template: Path to the Jinja2 template file to render.
+    :param outfile: Path to the file to write the rendered Jinja2 template to.
+    :param log: A logger.
+    :param dry_run: Run in dry-run mode?
+    :param values_needed: Just issue a report about variables needed to render the template? :raises
+        ValueError if values needed to render the template are missing.
+    """
+    cfg = _set_up_config_obj(config_file=config_file, key_eq_val_pairs=key_eq_val_pairs, log=log)
     template = J2Template(cfg, input_template, log_name=log.name)
-
     undeclared_variables = template.undeclared_variables
 
+    # If a report of variables required to render the template was requested, make that report and
+    # then return.
+
     if values_needed:
-        # Gather the undefined template variables.
-        log.info("Values needed for this template are:")
+        log.info("Values needed to render this template are:")
         for var in sorted(undeclared_variables):
             log.info(var)
         return
 
-    # Check for missing values.
+    # Check for missing values required to render the template. If found, report them and abort.
+
     missing = []
     for var in undeclared_variables:
         if var not in cfg.keys():
             missing.append(var)
-
     if missing:
-        log.critical("ERROR: Template requires variables that are not provided")
-        for key in missing:
-            log.critical(f"  {key}")
-        msg = "Missing values needed by template"
+        msg = "Template requires values that were not provided:"
         log.critical(msg)
+        for key in missing:
+            log.critical(key)
         raise ValueError(msg)
 
+    # In dry-run mode, display the rendered template and then return.
+
     if dry_run:
-        # Apply switch to allow user to view the results of rendered template
-        # instead of writing to disk. Render the template with the specified
-        # config object.
         rendered_template = template.render_template()
-        log.info(rendered_template)
-    else:
-        # Write out rendered template to file.
-        template.dump_file(outfile)
+        for line in rendered_template.split("\n"):
+            log.info(line)
```

Comment:
Log each line separately so that either all lines or none will have a logging prefix, which makes the visual output more consistent, and testing possible (as otherwise it's not clear whether to expect and account for log prefixes when looking for a specific expected line in the output).